tests/handlers/test_send_email.py

```diff
@@ -128,24 +164,32 @@ def test_send_email(self) -> None:
     )
     def test_send_email_force_tls(self) -> None:
         """Happy-path test that we can send email to an Implicit TLS server."""
+        # Patch ESMTPTLSClientFactory
+        synapse.handlers.send_email.ESMTPTLSClientFactory = TestingESMTPTLSClientFactory  # type: ignore[assignment]
```

Comment:
I think it would make more sense to actually use a [`patch`](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.patch) here, we wouldn't need to mess with the teardown at all then.

Reply:
Nice! Good one, it also helps remove the `setUp/tearDown` brainfart, can confirm it works, thank you!